src/Tools/dotnet-openapi/src/Commands/RemoveCommand.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Build.Evaluation;
+using Microsoft.Extensions.CommandLineUtils;
+using Microsoft.Extensions.Tools.Internal;
+
+namespace Microsoft.DotNet.OpenApi.Commands
+{
+    internal class RemoveCommand : BaseCommand
+    {
+        private const string CommandName = "remove";
+
+        public RemoveCommand(Application parent) : base(parent, CommandName)
+        {
+            _sourceProjectArg = Argument(SourceProjectArgName, $"The openapi reference to remove. Must represent a reference which is already in this project", multipleValues: true);
+        }
+
+        internal readonly CommandArgument _sourceProjectArg;
+
+        protected override Task<int> ExecuteCoreAsync()
+        {
+            var projectFile = ResolveProjectFile(ProjectFileOption);
+
+            var sourceFile = Ensure.NotNullOrEmpty(_sourceProjectArg.Value, SourceProjectArgName);
+
+            if (IsProjectFile(sourceFile))
+            {
+                RemoveServiceReference(OpenApiProjectReference, projectFile, sourceFile);
+            }
+            else
+            {
+                RemoveServiceReference(OpenApiReference, projectFile, sourceFile);
+
+                if (!Path.IsPathRooted(sourceFile))
+                {
+                    sourceFile = Path.Combine(WorkingDirectory, sourceFile);
+                }
+                File.Delete(sourceFile);
+            }
+
+            return Task.FromResult(0);
+        }
+
+        private void RemoveServiceReference(string tagName, FileInfo projectFile, string sourceFile)
+        {
+            var project = LoadProject(projectFile);
```

Comment:
Curious if you have tried this with multiple arguments? I thought loading the same project multiple times will fail since the static global project collection can only contain one copy of the project.

Reply:
I have tried it with multiple arguments (now added a test). As I recall it works for me because of the way I'm loading the projects (instead of just using the constructor.)